src/lando/main/scm/git.py

```diff
@@ -267,8 +274,17 @@ def update_repo(self, pull_path: str, target_cset: Optional[str] = None) -> str:
 
         This method uses the Git commands to update the repository
         located at the given pull path to the specified target changeset.
+
+        A new work branch will be created, using the current date in its name.
         """
-        branch = target_cset or self.default_branch
+        if not target_cset:
+            target_cset = self.default_branch
+
+        remote_branch = f"origin/{target_cset}"
+        if self._git_run("branch", "--list", "--remote", remote_branch, cwd=self.path):
+            # If the branch exists remotely, make sure we get the up-to-date version.
+            target_cset = remote_branch
```

Comment:
Curious why we are going with that `target_cset` terminology instead of a more git-centric one?

Reply:
(copying response to correct comment) Weight of history. It's what the standalone `HgRepo` object was using, and it got into the `abstract_scm`.

For git I'd use something like `base_commit`, but the meaning isn't changed.